### PR TITLE
Feature/form dropdown search 2

### PR DESCRIPTION
**Pull request by thomasowow**
Implements `FormDropdownSearch` instead of implementation of search in `FormDropdown`.

`FormDropdown` has basically been put back to its original form.

- [ ] Update PR with exporting new components and composables.

### CODE REVIEW

File: src/components/FormDropdownSearch/FormDropdownSearch.vue [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+        ? searchValueCached.value
+        : searchValue.value,
+    set: s => {
+        searchValueCached.value = '';
+
+        searchValue.value = s;
+
+        modelValue.value = undefined;
+    },
+});
```
Comment (albertpratomo) [inline src/components/FormDropdownSearch/FormDropdownSearch.vue:91]: Add comments? A bit hard to understand the why

Comment (albertpratomo) [reply src/components/FormDropdownSearch/FormDropdownSearch.vue:91]: Or how it works exactly

**Review by albertpratomo — CHANGES_REQUESTED** [review]

File: src/components/FormDropdownSearch/FormDropdownSearch.vue (hunk 1 above)
Comment (albertpratomo) [reply src/components/FormDropdownSearch/FormDropdownSearch.vue:91]: I mean the caching mechanism

File: src/components/FormDropdownSearch/FormDropdownSearch.test.ts [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    });
+
+    it('does not set searchValueCached to searchValue when searchValueCached has a value', () => {
+        const wrapper = mount(FormDropdownSearch);
+
+        wrapper.vm.searchValueCached = 'hello';
+
+        wrapper.vm.searchValue = 'do not copy me';
+
+        wrapper.vm.onFocus();
```
[the remark this answers is not in the document]
Comment (thomasowow) [reply src/components/FormDropdownSearch/FormDropdownSearch.test.ts:199]: To me it does not really help. Sometimes we just need both `wrapper` and `wrapper.vm`

**Pull request merged**